core/network/impl/peer_server_libp2p.cpp

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "network/impl/peer_server_libp2p.hpp"
+
+#include "libp2p/connection/stream.hpp"
+#include "network/impl/common.hpp"
+#include "scale/scale.hpp"
+
+namespace kagome::network {
+  using libp2p::basic::MessageReadWriter;
+
+  PeerServerLibp2p::PeerServerLibp2p(libp2p::Host &host,
+                                     libp2p::peer::PeerInfo peer_info,
+                                     common::Logger logger)
+      : host_{host},
+        peer_info_{std::move(peer_info)},
+        log_{std::move(logger)} {}
+
+  void PeerServerLibp2p::start() {
+    host_.setProtocolHandler(kSyncProtocol,
+                             [self{shared_from_this()}](auto stream) mutable {
+                               self->handleSyncProto(std::move(stream));
+                             });
+    host_.setProtocolHandler(
+        kGossipProtocol, [self{shared_from_this()}](auto stream) mutable {
+          self->handleGossipProto(std::make_shared<MessageReadWriter>(stream),
+                                  stream);
+        });
+  }
+
+  void PeerServerLibp2p::onBlocksRequest(BlocksRequestHandler handler) {
```

Comment:
usually function name onSomething() means handle event, not set handler.
setBlocksRequestHandler(handler) would sound more intuitive and self-describing.